### PR TITLE
Fixed security requirement resolution 

### LINKED ISSUE #222: openapi, Security parse issue: openApiSecurityScheme.Flows will always be null.
for example, parse below content of security,  the flows will be always empty. And in the v1.0.0-beta012 it works, but in v1.0.0-beta014 it was broken.

```yaml
components:
  securitySchemes:
    azure_auth: 
      type: oauth2
      description: Azure Active Directory OAuth2 Flow.
      in: header
      flows: 
        implicit:
          authorizationUrl: https://example.com/api/oauth/dialog
          tokenUrl: https://example.com/token
          scopes:
            user_impersonation: modify pets in your account
            read: read your pets
        authorizationCode:
          authorizationUrl: https://example.com/api/oauth/dialogcode
          tokenUrl: https://example.com/api/oauth/tokencode
          scopes:
            user_impersonation: modify pets in your account code.
            read: read your pets code.
```

### PR DESCRIPTION
**Pull request by darrelmiller**
Fixes #222

### CODE REVIEW

File: src/Microsoft.OpenApi/Services/OpenApiWalker.cs [hunk 1]
```diff
@@ -470,10 +470,32 @@ internal void Walk(OpenApiOperation operation)
             Walk(OpenApiConstants.Responses, () => Walk(operation.Responses));
             Walk(OpenApiConstants.Callbacks, () => Walk(operation.Callbacks));
             Walk(OpenApiConstants.Tags, () => Walk(operation.Tags));
-
+            Walk(OpenApiConstants.Security, () => Walk(operation.Security));
```
Comment (xuzhg) [inline src/Microsoft.OpenApi/Services/OpenApiWalker.cs]: If we miss add the process code, it should not be failed. So, I can't understand why? Is there any default setting that we can fix it?

**Comment by xuzhg** [comment]
Also, I'd like we should have some test cases to cover it.

File: src/Microsoft.OpenApi/Services/OpenApiWalker.cs (hunk 1 above)
Comment (darrelmiller) [reply src/Microsoft.OpenApi/Services/OpenApiWalker.cs]: I'm not sure I follow.  If we are missing Walk() methods then there may be references that remain unresolved or objects not validated.  Technically that's not an error as Unresolved references is a valid state.  They could be resolved manually later.

Yes, there has been an unfortunate number of cases of missing resolved references.  In theory these should have been picked up by regular test cases as unresolved references have null/default property values..  Not quite sure why more were not.

File: src/Microsoft.OpenApi.Readers/Services/OpenApiReferenceResolver.cs [hunk 2]
```diff
@@ -89,6 +89,23 @@ public override void Visit(OpenApiResponses responses)
             ResolveMap(responses);
         }
 
+        /// <summary>
+        /// Resolve all references to SecuritySchemes
```
Comment (xuzhg) [inline src/Microsoft.OpenApi.Readers/Services/OpenApiReferenceResolver.cs]: the comments should be "SecurityRequirement" not "SecuritySchemes" ?

Comment (darrelmiller) [reply src/Microsoft.OpenApi.Readers/Services/OpenApiReferenceResolver.cs]: Actually no.  SecurityRequirement is a dictionary whose keys are SecuritySchemes.  It is those keys that are being replaced if they are unresolved references.

**Pull request merged**